web-animations-2/Overview.bs

```diff
@@ -1096,8 +1096,9 @@ with:
 >     <em>and</em> <em>either</em> of the following conditions are met:
 >     1.  the <a>local time</a> is less than the <a>before-active boundary
 >         time</a>, <em>or</em>
->     1.  the <a>animation direction</a> is "backwards" and the
->         <a>local time</a> is equal to the <a>before-active boundary time</a>.
+>     1.  [=timeline duration=] is unresolved, the <a>animation direction</a>
+>         is "backwards" and the <a>local time</a> is equal to the
+>         <a>before-active boundary time</a>.
```

Comment:
This behavior of making the endpoints inclusive when timeline duration is resolved is a strange special case to have. This will also introduce many of the same edge cases from #6673 even though they will be limited to the endpoint of one animation, e.g.:

```js
element.animate({transform: ['translateY(0px)', 'translateY(10px)']}, {duration: CSS.percent(10), timeline: scrollTimeline, composite: 'add'}); 
element.animate({transform: ['translateY(10px)', 'translateY(50px)']}, {duration: CSS.percent(10), startDelay: CSS.percent(10), timeline: scrollTimeline, composite: 'add'})
```

At time 10% both effects will be active resulting in a transform of 20px. The same thing will happen for back to back effects in a SequenceEffect.

Reply:
FWIW, I do like the approach of making the active phase endpoint inclusive more than the auto fill mode, it just needs to only apply when the effect reaches the end of the timeline.

Reply:
Perhaps the simplest way to address this is calculate the effective timeline time by undoing the current time calculation, e.g.
effective timeline time = (local time / playback rate) + start_time

And then only use the endpoint inclusive condition if the effective timeline time is 0 (and backwards) or 100% (and forwards)

Reply:
It appears that this reverse engineering of effective timeilne time is not quite right for sequence effects since local time resets to zero at the start of each effect in the sequence.  Can likely come up with an equation based on inherited time to determine if aligned with the boundary of the timeline.  Just to be clear, are we only applying if scrollOffset = [0%, 100%] (or auto) and not at an arbitrary effective start or end offset?

Reply:
local_time is zero, but the start_time would be non-zero wouldn't it? Of course there may be simpler approaches that make it less likely we'll have to deal with equality of values.

Dealing with the actual max scroll offset is a good question, I think ideally we'd only apply this for the 100% / auto case so that stringing together animations on separate scroll timelines which stop when the next one starts also doesn't result in any overlap.

Reply:
Inferring the timeline time based on the inherited time and start time is only strictly correct for a running animation.  If the animation is paused we cannot tell if there was any lead/lag between the timeline and animation currentTime at the time of the pause.  Added an issue highlighting the problem. This discrepancy would only trigger an unexpected discrepancy for edge cases such as setting current time when pausing a progress-based animation precisely at the timeline boundary.  I don't believe these potential edge cases will be frequently triggered.